src/watchers/console.ts

```diff
@@ -0,0 +1,53 @@
+import * as fs from "fs";
+import {Watcher} from "./watcher";
+import {ID} from "../runtime/runtime";
+
+class consoleWatcher extends Watcher {
+
+    setup() {
+        if(console) {
+            let {program:me} = this;
+
+            me.watch("print to console log", ({find, record}) => {
+                let log = find("console/log");
+                return [
+                    record({log, text: log.text})
+                ]
+            })
+            me.asObjects<{log:ID, text:string}>(({adds, removes}) => {
```

Comment:
asDiffs is a fair bit faster than asObjects. For cases where you only need an id and a single attribute, I recommend using asDiffs instead. E.g.

```
me
  .watch("print to console log", ({find, record}) => {
    let log = find("console/log");
    return [log.add("text", log.text)];
  })
  .asDiffs(({adds}) => {
    for(let [log, _, text] of adds) {
      console.log(text);
    }
  })
```